LEGO1/lego/legoomni/src/build/legocarbuildpresenter.cpp

```diff
@@ -65,11 +71,103 @@ void LegoCarBuildAnimPresenter::ReadyTickle()
 	// TODO
 }
 
-// STUB: LEGO1 0x100789e0
-// STUB: BETA10 0x10070cdd
+// FUNCTION: LEGO1 0x100789e0
+// FUNCTION: BETA10 0x10070cdd
 void LegoCarBuildAnimPresenter::StreamingTickle()
 {
-	// TODO
+	if (!m_unk0x140->GetROI()) {
+		return;
+	}
+
+	m_mainSourceId = new LegoChar[strlen(m_action->GetAtomId().GetInternal()) + 1];
+	assert(m_mainSourceId);
+
+	strcpy(m_mainSourceId, m_action->GetAtomId().GetInternal());
+	m_mainSourceId[strlen(m_mainSourceId) - 1] = 'M';
+
+	FUN_10079160();
+
+	if (GameState()->GetCurrentAct() == LegoGameState::e_act2) {
+		m_unk0xc0 = 10;
+	}
+
+	MxS16 i;
+
+	for (i = 0; i < m_unk0xbe; i++) {
+		if (m_unk0xc0 == i) {
+			FUN_10079680(m_unk0x128[i].m_unk0x04);
+		}
+		else {
+			FUN_100795d0(m_unk0x128[i].m_unk0x04);
+		}
+
+		if (i < m_unk0xc0) {
+			FUN_10079050(i);
+			FUN_10079680(m_unk0x128[i].m_unk0x00);
+		}
+
+		LegoChar* name = m_unk0x128[i].m_unk0x04;
+
+		if (name) {
+			for (MxS32 j = 0; j <= m_roiMapSize; j++) {
+				LegoROI* roi = m_roiMap[j];
+
+				if (roi && roi->GetName() && (strcmpi(name, roi->GetName()) == 0)) {
+					roi->FUN_100a9dd0();
+					roi->FUN_100a9350("lego red");
+				}
+			}
+		}
+	}
+
+	LegoVideoManager* videoManager = VideoManager();
+	assert(videoManager); // verifies variable name 'videoManager'
+
+	Lego3DView* lego3dview = videoManager->Get3DManager()->GetLego3DView();
+	LegoROI* videoManagerROI = videoManager->GetViewROI();
+	LegoROI* local60 = m_unk0x140->GetROI();
+	LegoROI* camera = NULL;
+	MxFloat fov;
+
+	MxS16 totalNodes = CountTotalTreeNodes(m_anim->GetRoot());
+
+	for (i = 0; i < totalNodes; i++) {
+		LegoAnimNodeData* animNodeData = (LegoAnimNodeData*) GetTreeNode(m_anim->GetRoot(), i)->GetData();
+
+		if (strnicmp(animNodeData->GetName(), "CAM", strlen("CAM")) == 0) {
+			camera = local60->FindChildROI(animNodeData->GetName(), local60);
+			fov = atof(&animNodeData->GetName()[strlen(animNodeData->GetName()) - 2]);
+			break;
+		}
+	}
+
+	assert(camera); // verifies variable name 'camera'
+
+	LegoROI* targetROI = local60->FindChildROI("TARGET", local60);
+
+	Mx3DPointFloat dirVec;
+
+	Vector3 cameraPosition(camera->GetWorldPosition());
+	Vector3 upVec(camera->GetWorldUp());
+	Vector3 targetPosition(targetROI->GetWorldPosition());
+
+	MxMatrix localTransform;
+
+	dirVec[0] = targetPosition[0] - cameraPosition[0];
```

Comment:
this is probably the `VMV3` macro in `vec.h`

Reply:
I thought that too and tried it, but it did decrease the match for LEGO1. I could still change if if desired.